src/main/java/uk/gov/hmcts/cmc/claimstore/services/FormaliseResponseAcceptanceService.java

```diff
@@ -0,0 +1,185 @@
+                String contentBySetDate = prepareOfferContentsBySetDate(response, builder, completionDate);
+                builder.content(contentBySetDate);
+                break;
+            case INSTALMENTS:
+                RepaymentPlan repaymentPlan = paymentIntention.getRepaymentPlan().orElseThrow(IllegalAccessError::new);
+                builder.completionDate(repaymentPlan.getCompletionDate());
+                builder.content(String.format(
+                    "%s will pay instalments of %s %s. The first instalment will be paid by %s.",
+                    response.getDefendant().getName(),
+                    formatMoney(repaymentPlan.getInstalmentAmount()),
+                    repaymentPlan.getPaymentSchedule().getDescription(),
+                    formatDate(repaymentPlan.getFirstPaymentDate())
+
+                ));
+                break;
+            default:
+                throw new IllegalStateException("Invalid payment option " + paymentIntention.getPaymentOption());
+        }
+        return builder.build();
+    }
+
+    private String prepareOfferContentsBySetDate(
+        Response response,
+        Offer.OfferBuilder builder,
+        LocalDate completionDate
+    ) {
+        builder.completionDate(completionDate);
+        String amount;
+        switch (response.getResponseType()) {
+            case PART_ADMISSION:
+                PartAdmissionResponse partAdmissionResponse = (PartAdmissionResponse) response;
+                amount = formatMoney(partAdmissionResponse.getAmount());
+                break;
+            case FULL_ADMISSION:
+                amount = "the full amount";
+                break;
+            default:
+                throw new IllegalStateException("Invalid response type " + response.getResponseType());
+        }
+        return String.format("%s will pay %s, no later than %s",
+            response.getDefendant().getName(), amount, formatDate(completionDate)
+        );
+    }
+
+    private void formaliseCCJ(Claim claim, ResponseAcceptation responseAcceptation, String authorisation) {
+        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+        PaymentIntention acceptedPaymentIntention = acceptedPaymentIntention(responseAcceptation, response);
+
+        CountyCourtJudgment countyCourtJudgment = CountyCourtJudgment.builder()
+            .defendantDateOfBirth(defendantDateOfBirth(response.getDefendant()))
+            .paymentOption(acceptedPaymentIntention.getPaymentOption())
+            .paidAmount(responseAcceptation.getAmountPaid())
+            .repaymentPlan(acceptedPaymentIntention.getRepaymentPlan().orElse(null))
+            .payBySetDate(acceptedPaymentIntention.getPaymentDate().orElse(null))
+            .build();
+
+        this.countyCourtJudgmentService.save(
+            claim.getSubmitterId(),
+            countyCourtJudgment,
+            claim.getExternalId(),
+            authorisation,
+            true);
```

Comment:
ask question to Robert, if we have to pass false as we are changing ccd by admission to ccd by default?

Reply:
That is separate story to take care of this.